### PR TITLE
Add overview section to sync-endpoint

### PR DESCRIPTION
**Pull request by EstherPhebe**
addresses odk-x/tool-suite-X#207

#### What is included in this PR?
1. Add overview section to sync-endpoint

### CODE REVIEW

File: odkx-src/sync-endpoint.rst [hunk 1]
```diff
@@ -32,6 +33,49 @@ in a cloud-based virtual machine, or on your own infrastructure.
...
+
+Overview
+--------------
+
+.. _sync-endpoint-sync-protocol:
+
+ODK-X Synchronization Protocol
+"""""""""""""""""""""""""""""""""
+
+ODK-X’s synchronization protocol is based on a REST architecture that keeps the data on multiple devices synchronized to a master copy stored on the ODK-X Sync-Endpoint. All synchronization operations are idempotent, thereby simplifying the API because clients do not have to worry about losing data, as requests can be safely repeated in environments where network timeouts occur.
```
Comment (elmps2018) [inline odkx-src/sync-endpoint.rst:46]: I am not sure what idempotent means...

File: odkx-src/sync-endpoint.rst [hunk 2]
```diff
@@ -32,6 +33,49 @@ in a cloud-based virtual machine, or on your own infrastructure.
...
+Cell-based conflicts would be an even smaller unit of data that would further reduce conflicts; however, in a single row, the cell values are often interrelated. We felt that too much context could be lost by treating the cells separately, thus leading to reconciliation errors. 
+
+A conflict is defined as two users with different updates to the same row. ODK-X uses table locks on the server to ensure only a single change to a data row can occur at any time. When the :dfn:`runner-up` client finally obtains the lock and attempts to alter the same row, the update will be rejected as a conflict. Once a conflict is detected, the user manually determines which version of data is correct between their pending changes on the local client and the updated data row on the server. 
+The rationale for having the user who caused the conflict also resolve the conflict is that the user was recently working with data and is likely to have the necessary information and context on how best to resolve the conflict.
+
+You can learn more here: :doc:`odk-2-sync-protocol`
+
+.. _sync-endpoint-server-tech:
+
+ODK-X Sync Endpoint Server Technologies
```
Comment (elmps2018) [inline odkx-src/sync-endpoint.rst:60]: I would switch the order to the overview sections -- put this first and then the details on conflicts second.

File: odkx-src/sync-endpoint.rst [hunk 3]
```diff
@@ -32,6 +33,49 @@ in a cloud-based virtual machine, or on your own infrastructure.
...
+
+To direct the incoming web request, the Sync-Endpoint uses nginx to route the web request to the proper microservice able to properly respond to the request. The central microservice of the Sync-Endpoint is a Java web application that provides the ODK-X REST synchronization protocol. 
+
+.. figure:: /img/sync-endpoint/endpoint-docker-swarm.*
+   :alt: An architecture diagram of the six main microservices running in a Docker swarm
+  
+   An architecture diagram of the six main microservices running in a Docker swarm. The six main microservices are: nginx, Sync-Endpoint REST Interface, Sync-Endpoint Web UI, PostgreSQL, phpLDAPadmin, and OpenLDAP.
+
+The REST protocol microservice runs an Apache Tomcat webserver. By default, the Sync-Endpoint uses a PostgreSQL server running as a microservice; however, the endpoint is designed to integrate with other databases (e.g., MySQL).
+The Sync-Endpoint Web UI is the microservice that provides the user interface for the sync-endpoint server. Sync-Endpoint is also a Java web application running inside an Apache Tomcat webserver.
```
Comment (elmps2018) [inline odkx-src/sync-endpoint.rst:73]: Is the lat sentence repeated? It seems similar to the REST protocol.

**Review by elmps2018 — COMMENTED** [review]
This is a great addition! A few specific comments.

**Review by elmps2018 — DISMISSED** [review]

File: odkx-src/sync-endpoint.rst [hunk 4]
```diff
@@ -32,6 +33,48 @@ in a cloud-based virtual machine, or on your own infrastructure.
...
+
+ODK-X’s synchronization protocol is based on a REST architecture that keeps the data on multiple devices synchronized to a master copy stored on the ODK-X Sync-Endpoint. Clients do not have to worry about losing data, as API requests can be safely repeated in environments where network timeouts occur.
+
+To minimize data updates that conflict, data updates are processed as row-based changes to keep changes small. For example, when performing a cold chain inventory, if updates were at a coarse granularity, such as table-based or file-based, a conflict might be detected for two workers updating refrigerators while working at different sites. 
+By keeping conflict detection at the row-level, multiple users can make updates to shared data tables, and the system will detect that there is not a conflict as long as the same row is not updated by different users between their synchronizations. 
+
+Cell-based conflicts would be an even smaller unit of data that would further reduce conflicts; however, in a single row, the cell values are often interrelated. We felt that too much context could be lost by treating the cells separately, thus leading to reconciliation errors.
```
Comment (wbrunette) [inline odkx-src/sync-endpoint.rst:71]: I am not sure we should talk about cell-based conflicts as we don't do it and could add to confusion. @elmps2018 thoughts?

**Review by wbrunette — CHANGES_REQUESTED** [review]

File: odkx-src/sync-endpoint.rst (hunk 4 above)
Comment (elmps2018) [reply odkx-src/sync-endpoint.rst:71]: Can cut two sentences on cell-based conflicts.

**Review by wbrunette — APPROVED** [review]

**Pull request merged**